### PR TITLE
Display the user data in the edit form.

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Load and display user data in the edit form. Fixes: https://github.com/canonical-web-and-design/maas-squad/issues/1356.
- See a password length. Fixes: #47.
- Update the Formik fields to reduce duplication and be testable.
- Set unique ids on form fields. Fixes: #72

QA:
- Navigate to the user list.
- Click on the edit icon for a user.
- You should get the edit form with the user's data.

### CODE REVIEW

**Comment by squidsoup** [comment]
Question from QA, is this tickbox behaving correctly? 

image

**Comment by squidsoup** [comment]
Might need to get some design feedback on this form, the Change Password link looks like it could use some padding or something? 🤷‍♂ 

image

**Comment by huwshimi** [comment]
@squidsoup that help text shouldn't have been there, that was a bug in my code, it matches the design now.

File: ui/src/app/settings/containers/UserEdit/UserEdit.js [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
 
 export const UserEdit = () => {
-  return <UserForm title="Edit user" />;
+  const { id } = useParams();
+  const loading = useSelector(selectors.users.loading);
+  const user = useSelector(state =>
+    selectors.users.getById(state, parseInt(id))
+  );
+  const loaded = useFetchOnce(actions.users.fetch, selectors.users.loaded);
+  if (loading || !loaded) {
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/settings/containers/UserEdit/UserEdit.js]: For some reason it was breaking if I didn't have that, but I can't reproduce now. Maybe it was something else. I'll remove the check.

File: ui/src/app/settings/components/UserForm/UserForm.js [hunk 2]
```diff
@@ -30,16 +28,25 @@ const UserSchema = Yup.object().shape({
...
+
+const UserSchema = Yup.object().shape({
+  ...schemaFields,
+  password: schemaFields.password.required("Password is required")
 });
 
+const UserEditSchema = Yup.object().shape(schemaFields);
+
 export const UserForm = ({ title, user }) => {
+  const editing = !!user;
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/settings/components/UserForm/UserForm.js]: You don't want to use `!!`?

File: ui/src/app/settings/components/UserForm/UserForm.js [hunk 3]
```diff
@@ -49,108 +56,30 @@ export const UserForm = ({ title, user }) => {
               passwordConfirm: "",
               username: user ? user.username : ""
             }}
-            validationSchema={UserSchema}
+            validationSchema={editing ? UserEditSchema : UserSchema}
             onSubmit={(values, { setSubmitting }) => {
               setTimeout(() => {
-                alert(JSON.stringify(values, null, 2));
+                const nameParts = values.fullName && values.fullName.split(" ");
```
[the remark this answers is not in the document]
Comment (huwshimi) [reply ui/src/app/settings/components/UserForm/UserForm.js]: Smart.

**Pull request merged**